compose/config/config.py

```diff
@@ -165,13 +164,11 @@ def find_candidates_in_parent_dirs(filenames, path):
     return (candidates, path)
 
 
-@validate_top_level_object
-@validate_service_names
```

Comment:
You've removed `validate_service_names` without putting it back in elsewhere. Was this on purpose? You still need this validation to happen, validating a `top_level_object` as a dict on it's own isn't enough.